daemon/resize.go

```diff
@@ -37,5 +37,15 @@ func (daemon *Daemon) ContainerExecResize(name string, height, width int) error
 	if err != nil {
 		return err
 	}
+	// Without the lock, resize would race with exec start when the start handler is still in progress.
```

Comment:
If think it could simply be:

```
Synchronize with exec-start to ensure the resize request is not sent before the process is actually started.
```

Would that work for you too @tonistiigi ?